src/rubyLsp.ts

```diff
@@ -0,0 +1,401 @@
+import path from "path";
+
+import * as vscode from "vscode";
+import { Range } from "vscode-languageclient/node";
+
+import { Telemetry } from "./telemetry";
+import DocumentProvider from "./documentProvider";
+import { Workspace } from "./workspace";
+import { Command, STATUS_EMITTER, pathExists } from "./common";
+import { VersionManager } from "./ruby";
+import { StatusItems } from "./status";
+import { TestController } from "./testController";
+import { Debugger } from "./debugger";
+
+// The RubyLsp class represents an instance of the entire extension. This should only be instantiated once at the
+// activation event. This class controls all of the existing workspaces, telemetry and handles all commands
```

Comment:
Does one RubyLsp instance control multiple workspaces, or it's mapped 1 to 1? I think this can be noted in the comment here too.

Reply:
It controls all of the workspaces. There's only one `RubyLsp` instance, but many `Workspace` instances. I updated the comment, please let me know if it's clear or if I can better word it.